ThiefGame/Assets/Scripts/UI/TileDeckUI.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.Design.Serialization;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+public class DeckUI : MonoBehaviour
+{
+    [SerializeField] private UIDocument document;
+    [SerializeField] private StyleSheet styleSheet;
+
+    [SerializeField] private TileDeckController tileDeckController;
+
+    public static event Action<int> TileFromHandSelected;
+    public static event Action DiscardButtonPressed;
+
+    private List<BaseTile> _currentHandTiles;
+    private Coroutine _currentGenerateRoutine;
+
+    private void Awake()
+    {
+        _currentHandTiles = new();
+    }
+
+    private void Start()
+    {
+        //Using a Coroutine because of the abilitity to do animations in the future
```

Comment:
Good call! I think using a Coroutine from the start to anticipate it getting more complex if a great call